automation_tools/satellite6/upgrade/tests/utils.py

```diff
@@ -0,0 +1,152 @@
+    delete_rhevm_instance,
+    get_rhevm_client,
+    remove_all_docker_containers,
+    wait_till_rhev_instance_status
+)
+from automation_tools.satellite6.upgrade.tasks import docker_execute_command
+from automation_tools.satellite6.upgrade.tools import logger
+from fabric.api import execute, run
+
+rpm1 = 'https://inecas.fedorapeople.org/fakerepos/zoo3/bear-4.1-1.noarch.rpm'
+rpm2 = 'https://inecas.fedorapeople.org/fakerepos/zoo3/camel-0.1-1.noarch.rpm'
+data = {}
+rhev_template = 'sat6-docker-upgrade-template'
+
+logger = logger()
+
+
+def create_dict(dict):
+    data.update(dict)
+    print data
+    with open('{}.pkcl'.format('datastore'), 'wb') as pref:
+        pickle.dump(data, pref)
+
+
+def get_dict():
+    print data
+
+
+def generate_satellite_docker_clients_on_rhevm_upgrade(
+        client_os, clients_count, ak):
+    """Generates satellite clients on docker as containers
+
+    :param string client_os: Client OS of which client to be generated
+        e.g: rhel6, rhel7
+    :param string clients_count: No of clients to generate
+    :param string ak : Activation key name, to register clients
+
+    Environment Variables:
+
+    RHEV_SAT_HOST
+        The satellite hostname for which clients to be generated and
+        registered
+    RHEV_CLIENT_AK
+        The AK using which client will be registered to satellite
+    """
+    if int(clients_count) == 0:
+        logger.warning(
+            'Clients count to generate on Docker should be atleast 1 !')
+        sys.exit(1)
+    satellite_hostname = get_satellite_host()  # specify satellite hostname
+    result = {}
+    for count in range(int(clients_count)):
+        if bz_bug_is_open('1405085'):
+            time.sleep(5)
+        hostname = '{0}dockerclient{1}'.format(count, client_os)
+        container_id = run(
+            'docker run -d -h {0} -e "SATHOST={1}" '
+            '-e "AK={2}" upgrade:{3}'.format(
+                hostname, satellite_hostname, ak, client_os))
+        result[hostname] = container_id
+    return result
+
+
+def get_id(entity, name, org_id):
+    result = hammer_info_via_name(entity, name, org_id)
+    return get_attribute_value(result, name, 'id')
+
+
+def get_latest_repo_version(cv_name):
+    cv_version_data = hammer(
+        'content-view version list --content-view {} '
+        '--organization-id 1'.format(cv_name))
+    latest_cv_ver = sorted([float(data['name'].split(
+        '{} '.format(cv_name))[1]) for data in cv_version_data]).pop()
+    cv_ver_id = get_attribute_value(cv_version_data, '{0} {1}'.format(
+        cv_name, latest_cv_ver), 'id')
+    return cv_ver_id
+
+
+def create_vm():
+    usecase_instance = rhev_template + 'usecase-testing'
+    delete_rhevm_instance(usecase_instance, timeout=5)
+    create_rhevm_instance(usecase_instance,
+                          rhev_template,
+                          'SAT-QE',
+                          'SAT-QE',
+                          'SAT-QE')
+
+
+def dockerize(ak_name):
```

Comment:
Not sure about what this function does, it deals with a pre created docker VM and then do some rhvm actions. I think a docstring would clarify that and maybe some more descriptive name.